ibis/pandas/execution/window.py

```diff
@@ -236,6 +288,11 @@ def execute_window_op(
     assert len(data) == len(
         series
     ), 'input data source and computed column do not have the same length'
+
+    # trim data to original time context
+    if timecontext:
+        series = trim_with_timecontext(series, timecontext)
```

Comment:
Minor comments, we can make `trim_with_timecontext` noop if timecontext is None so we don't need the if statement here.

Reply:
Sure, I have moved this logic into `trim_with_timecontext` and made it a no-op.